discord-scripts/invite-management.ts

```diff
@@ -0,0 +1,181 @@
+import { Robot } from "hubot"
+import {
+  Client,
+  TextChannel,
+  VoiceChannel,
+  StageChannel,
+  ApplicationCommandOptionType,
+} from "discord.js"
+
+async function createInvite(
+  discordClient: Client,
+  channelId: string,
+  maxAge: number | string = 86400,
+  maxUses: number | string = 0,
+): Promise<string | null> {
+  if (!discordClient) {
+    throw new Error("Discord client is not initialized.")
+  }
+
+  let channelOrigin: TextChannel | VoiceChannel | StageChannel | null = null
+  try {
+    const fetchedChannel = await discordClient.channels.fetch(channelId)
+    if (
+      fetchedChannel instanceof TextChannel ||
+      fetchedChannel instanceof VoiceChannel ||
+      fetchedChannel instanceof StageChannel
+    ) {
+      channelOrigin = fetchedChannel
+    }
+  } catch (error) {
+    throw new Error("Channel not found or access denied.")
+  }
+
+  if (!channelOrigin) {
+    return null
+  }
+  const numericMaxAge = parseInt(maxAge.toString(), 10)
+  const numericMaxUses = parseInt(maxUses.toString(), 10)
+  if (Number.isNaN(numericMaxAge) || Number.isNaN(numericMaxUses)) {
+    throw new Error("maxAge and maxUses must be valid numbers.")
+  }
```

Comment:
Even if we did allow these from the command, it's generally a good idea to keep the parsing of string inputs as close to the receiving of those inputs as possible, so that this function can stay focused exclusively on invite creation rather than input processing.